### PR TITLE
FE: do not enhance `?` in jspecify `NullMarked` scope

### PR DESCRIPTION
**Pull request by pyos**
No clue whether this should be done for `@TypeQualifierDefault` since it appears to have no specification whatsoever.

 #KT-48262 Fixed

### CODE REVIEW

**Review by petukhovv — APPROVED** [review]

**Comment by pyos** [comment]
@petukhovv I see you've cherry-picked this PR into 1.5.30, but will you merge it into master as well?

**Comment by petukhovv** [comment]
Yes, of course. I'm waiting for rr, and after passing the tests will merge it into master.

**Pull request merged**
